Kitodo/src/main/java/de/sub/goobi/export/dms/ExportDms.java

```diff
@@ -274,93 +289,88 @@ public boolean startExport(Process process, URI inZielVerzeichnis, DigitalDocume
-                gdzfile.write(userHome + File.separator + atsPpnBand + ".xml");
-            }
+    private void exportWithImport(Process process, FileformatInterface gdzfile, URI userHome)
+            throws IOException, PreferencesException, TypeNotAllowedForParentException, WriteException {
+        String fileFormat = process.getProject().getFileFormatDmsExport();
+        String processTitle = process.getTitle();
+        if (exportDmsTask != null) {
+            exportDmsTask.setWorkDetail(atsPpnBand + ".xml");
+        }
+        if (MetadataFormat.findFileFormatsHelperByName(fileFormat) == MetadataFormat.METS) {
+            // if METS, then write by writeMetsFile...
+            writeMetsFile(process, fileService.createResource(userHome, File.separator + atsPpnBand + ".xml"), gdzfile,
+                false);
+        } else {
+            // ...if not, just write a fileformat
+            gdzfile.write(userHome + File.separator + atsPpnBand + ".xml");
+        }
 
-            /* ggf. sollen im Export mets und rdf geschrieben werden */
-            if (MetadataFormat.findFileFormatsHelperByName(
-                process.getProject().getFileFormatDmsExport()) == MetadataFormat.METS_AND_RDF) {
-                writeMetsFile(process, fileService.createResource(userHome, File.separator + atsPpnBand + ".mets.xml"),
-                    gdzfile, false);
-            }
+        // if necessary, METS and RDF should be written in the export
+        if (MetadataFormat.findFileFormatsHelperByName(fileFormat) == MetadataFormat.METS_AND_RDF) {
+            writeMetsFile(process, fileService.createResource(userHome, File.separator + atsPpnBand + ".mets.xml"),
+                gdzfile, false);
+        }
 
-            Helper.setMeldung(null, process.getTitle() + ": ", "DMS-Export started");
-            if (!ConfigCore.getBooleanParameter("exportWithoutTimeLimit")) {
-                DmsImportThread agoraThread = new DmsImportThread(process, atsPpnBand);
-                agoraThread.start();
-                try {
-                    /* 30 Sekunden auf den Thread warten, evtl. killen */
-                    agoraThread.join(process.getProject().getDmsImportTimeOut().longValue());
-                    if (agoraThread.isAlive()) {
-                        agoraThread.stopThread();
-                    }
-                } catch (InterruptedException e) {
-                    if (exportDmsTask != null) {
-                        exportDmsTask.setException(e);
-                    } else {
-                        Helper.setFehlerMeldung(process.getTitle() + ": error on export - ", e.getMessage());
-                    }
-                    logger.error(process.getTitle() + ": error on export", e);
+        Helper.setMeldung(null, process.getTitle() + ": ", "DMS-Export started");
+        if (!ConfigCore.getBooleanParameter("exportWithoutTimeLimit")) {
+            DmsImportThread agoraThread = new DmsImportThread(process, atsPpnBand);
```

Comment:
This is an asynchronous task - the old name (I hate this name in 2.x too) is wrong and not helpful for any one. Please change it to f.e. `asyncThread`